runtime/src/bank.rs

```diff
@@ -808,6 +829,9 @@ pub struct Bank {
     epoch_reward_status: EpochRewardStatus,
 
     transaction_processor: TransactionBatchProcessor<BankForks>,
+
+    /// Collected fee details
+    collector_fee_details: RwLock<CollectorFeeDetails>,
```

Comment:
This is to replace original `collector_fee: u64` to store collected fees as bank progressing.

One difference to point out: original `collector_fees` is (de)serialized, which, as I understand, is not functionally necessary. Fees collected are distributed when bank is frozen; `collector_fees` of frozen (or rooted) bank loaded from ledger is no used, except helps to show how many fees that bank had collected.

Unless there's compelling reason, I feel it's better not to (de)serialize `collector_fee_details` to snapshot.